client/jetpack-cloud/sections/partner-portal/hooks.ts

```diff
@@ -38,3 +38,62 @@ export function useRecentPaymentMethodsQuery( { enabled = true }: UseQueryOption
 		}
 	);
 }
+
+/**
+ * Returns false; if flag ever becomes true, then returns true from then onwards.
+ *
+ * Useful to determine if pagination should be shown based on whether flag is or has been true at some point.
+ * For example, Stripe responses include a boolean `has_more` pagination value which you can pass as the
+ * argument and use the return value to conditionally render your pagination.
+ *
+ * @param {boolean} flag
+ * @returns {boolean}
+ */
+export function usePermanentFlag( flag: boolean ): boolean {
```

Comment:
We call it `usePermantFlag` which sounds very generic but it will "only" work for one state instance (in our case that instance is "has more" for pagination).

Should it perhaps be named so some more specific to "has more" / pagination or should we allow a specific state as a parameter, so we it will work more as a generic utility function? (I'm mostly saying this because the function lives in a mixed purpose hooks file and not a specific "Pagination hooks" file where the context could be assumed)

Reply:
It is generic - it's not tied to hasMore, it's just that hasMore is its only use *for now*.